packages/ra-core/src/controller/useRecordSelection.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react';
+// @ts-ignore
+import { useSelector, useDispatch } from 'react-redux';
+import { setListSelectedIds, toggleListItem } from '../actions/listActions';
+import { Identifier, ReduxState } from '../types';
+
+/**
+ * Get the list of selected items for a resource, and callbacks to change the selection
+ *
+ * @param resource The resource name, e.g. 'posts'
+ *
+ * @returns {Object} Destructure as [selectedIds, { select, toggle, clearSelection }].
+ */
+const useSelectItems = (resource: string) => {
+    const dispatch = useDispatch();
+    const selectedIds = useSelector(
+        (reduxState: ReduxState) =>
+            reduxState.admin.resources[resource].list.selectedIds,
+        [resource]
+    );
+    const selectionModifiers = {
+        selectedIds: useSelector(
```

Comment:
Why is this a modifier? Don't we already send it as the first item in the tupple?